### DISCUSSION TITLE
Creating a Pull Request - LGMSOC Participant(Amit)

### DISCUSSION THREAD

**Discussion by prathimacode-hub** [prathimacode-hub/Awesome_Python_Scripts#295] [Q&A]
I shall share you the easiest way to create a pull request without any hassles especially for beginners to open source and github.

**Comment by prathimacode-hub** [comment]
Process to Follow (without command line confusion)👍

Fork the repository. You are now in your profile repository.
You can see a "Add File" option in the local repository
You can add files or folders using the 2options given
Create new file option helps in creating both file and sub repository within this repository.
Example : You want to add up your "Event Handling" Script to the repository
Go to "GUIScripts" repository. Select "Create a new file" option,
It opens up like this - /Awesome_Python_Scripts/GUIScripts
To add up your repository and file you do :
/Awesome_Python_Scripts/GUIScripts/Event Handling/README.md
This add up would not only add 'Event Handling' as repository but also adds a empty 'README.md' file.
You can add your README content in 'README.md file' and commit the changes
Now you have created a repository of your desired script, you are yet to add your python program and related files.
From the very same "Add File" option, select "Upload files" and add your python program from your local drive of your system.
Make sure the script file name is "event_handling.py" before you upload the file to this repository.
Now, we needed to add images and screenshots of your project, so we again use " Create a New File" Option you do
/Awesome_Python_Scripts/GUIScripts/Event Handling/Images/README.md and commit it
You're now in "Images" folder. Add the relevant images and screenshots you want to add up in your README.md file of your project by using "Add File" Option from local drive of your system
Get back to your project repository of "Event Handling" and using "Create a New File" Option you do
/Awesome_Python_Scripts/GUIScripts/Event Handling/requirements.txt, add the imported libraries used in your program in this file and commit it
We haven't added our output screenshots or images in "README.md" of your project. Open up your README.md file. There you could see a "pencil button" that is to edit the file. You can add up the image links you had stored in "Images" folder in README file by copying the file path.
If we remember, we had created a README.md file in "Images" folder as well to get it registered as a repository.
You can now delete that file using "delete option that is trash button". And you are done.
Finally, You have successfully added up all your required files needed.
Now select "Contribute" option above and select Open Pull request, and add a defined title of your project, a proper commit message in a description using the PR template given.
Finally, "Create a Pull Request" option to synchronize your fork to the committed repository. That's it.
Voila, you did it. Congrats, on learning how to create a PR in a easier way. No much fuss around to deal with.

PR can be created without the use of command line terminal, but preferrably that way to show professionalism.
Contribution is all that matters, so start contributing and feel free to do more PR in future.

@AmitGupta700 , I like your confidence and enthusiasm to learn. Keep going.
If you found this helpful and easier to try, you can do create the PR of the same script with required changes I have mentioned in this comment.

Looking forward to merge the upcoming PR from you. Thank You.

**Reply by AmitGupta700** [reply]
Yes Ma'am it was very helpfull. Thank You!!

**Reply by prathimacode-hub** [reply]
> Yes Ma'am it was very helpfull. Thank You!!

Most welcome. Hopefully I get right PR from you very soon.

**Reply by AmitGupta700** [reply]
Ma'am, I have done please see it.
